impeller/core/host_buffer.h

```diff
@@ -116,14 +114,42 @@ class HostBuffer final : public std::enable_shared_from_this<HostBuffer>,
   void Reset();
 
   //----------------------------------------------------------------------------
-  /// @brief Returns the size of the HostBuffer in memory in bytes.
+  /// @brief Returns the capacity of the HostBuffer in memory in bytes.
   size_t GetSize() const;
 
+  //----------------------------------------------------------------------------
+  /// @brief Returns the size of the currently allocated HostBuffer memory in
+  /// bytes.
+  size_t GetLength() const;
+
  private:
-  mutable std::shared_ptr<DeviceBuffer> device_buffer_;
-  mutable size_t device_buffer_generation_ = 0u;
-  size_t generation_ = 1u;
-  std::string label_;
+  class HostBufferState : public Buffer, public Allocation {
+   public:
```

Comment:
Nit: mark as struct and remove access modifier?

Reply:
Done